cadc-vos-server/src/main/java/org/opencadc/vospace/server/pkg/VospacePackageRunner.java

```diff
@@ -237,35 +281,40 @@ protected URL getURL(VOSURI nodeURI)
     }
 
     // Get a PackageItem for a directory from the given ContainerNode.
-    protected PackageItem getDirectoryPackageItem(ContainerNode node) {
-        String nodePath = Utils.getPath(node);
-        log.debug("ContainerNode path: " + nodePath);
-
-        return new PackageItem(nodePath);
+    protected PackageItem getDirectoryPackageItem(ContainerNode node, String relativePath) {
+        return new PackageItem(relativePath);
     }
 
     // Get a PackageItem for a file from the given DataNode.
-    protected PackageItem getFilePackageItem(DataNode node)
+    protected PackageItem getFilePackageItem(DataNode node, String relativePath)
             throws NodeNotFoundException, PrivilegedActionException {
-        String nodePath = Utils.getPath(node);
-        log.debug("DataNode path: " + nodePath);
 
+        String nodeRelativePath = relativePath + "/" + node.getName();
+        String nodePath = Utils.getPath(node);
         VOSURI nodeURI = new VOSURI(resourceID, nodePath);
         URL endpointURL = getURL(nodeURI);
-        return new PackageItem(nodePath, endpointURL);
+        return new PackageItem(nodeRelativePath, endpointURL);
     }
 
     // Get a PackageItem for a symbolic link from the given LinkNode.
-    protected PackageItem getSymbolicLinkPackageItem(LinkNode node) {
+    protected PackageItem getSymbolicLinkPackageItem(LinkNode node, String relativePath) {
         String nodePath = Utils.getPath(node);
         log.debug("LinkNode path: " + nodePath);
 
-        URI targetURI = node.getTarget();
+        // check that the link target is in the package
+        // return a null PackageItem if it is out of the package
         VOSURI nodeURI = new VOSURI(this.resourceID, nodePath);
-        String linkPath = nodeURI.getPath();
-        log.debug("LinkNode target path: " + linkPath);
+        URI targetURI = node.getTarget();
+        boolean inPackage = isTargetInPackage(nodeURI.getURI(), targetURI, relativePath);
```

Comment:
Since this uses the "early return" strategy (I prefer that), it would be nicer if the utility method was something like 
```
boolean isExternal(LinkNode link, String packageBasePath)
```
so the usage is positive, like 
```
if (externalLink) {
   return null;
}
```
aside: note committed to `isExternal` because it might be too vague, maybe `isOutsidePackage`???

Secondarily, if there are multiple targets in the package job, that means there are multiple basePkgPath(s) which define internal vs external. I don't think we discussed that explicitly but as soon as I tried to write the above method signature is became obvious.

Reply:
`isExternalTarget`? Yes there can be multiple targets, and the relative path is the path to the target for the node.

Reply:
by multiple targets I meant multiple input URIs to the pkg job, eg
```
\/a
\/b
```
In that case, a link node somewhere under `\/a` that points under `\/b` *should* be included in  the package. So the package as a whole has multiple starting paths, not just one, and `isExternalTarget` needs to take that into account.

Reply:
So `isExternalTarget` has to check if the link target is under all the input targets. Okay, I'll see how that can be done.